modules/dnn/src/dnn.cpp

```diff
@@ -1403,6 +1424,10 @@ struct Net::Impl : public detail::NetImplBase
                   preferableTarget == DNN_TARGET_FPGA
             );
         }
+#endif
+#ifdef HAVE_WEBNN
+        CV_Assert(preferableBackend != DNN_BACKEND_WEBNN ||
+                  preferableTarget == DNN_TARGET_WEBNN);
```

Comment:
Should we assert like
```c++
if (preferableBackend == DNN_BACKEND_WEBNN) {
  CV_Assert(preferableTarget == DNN_TARGET_CPU || preferableTarget == DNN_TARGET_GPU)
}
```

Reply:
Please define `DNN_BACKEND_WEBNN` and `DNN_TARGET_GPU` in dnn.hpp.

Reply:
fixed